projects/plugins/jetpack/modules/carousel/jetpack-carousel.js

```diff
@@ -1606,45 +1606,54 @@
 				var grandparent = parent.parentElement;
 
 				// If Gallery is made up of individual Image blocks check for custom link before
-				// loading carousel.
+				// loading carousel. The custom link may be the parent or could be a descendant
+				// of the parent if the image has rounded corners.
+				var parentHref = null;
 				if ( grandparent && grandparent.classList.contains( 'wp-block-image' ) ) {
-					var parentHref = parent.getAttribute( 'href' );
-
-					// If the link does not point to the attachment or media file then assume Image has
-					// a custom link so don't load the carousel.
-					if (
-						parentHref.split( '?' )[ 0 ] !==
-							target.getAttribute( 'data-orig-file' ).split( '?' )[ 0 ] &&
-						parentHref !== target.getAttribute( 'data-permalink' )
-					) {
-						return;
-					}
+					parentHref = parent.getAttribute( 'href' );
+				} else if (
+					parent &&
+					parent.classList.contains( 'wp-block-image' ) &&
+					parent.querySelector( 'a' )
+				) {
+					parentHref = parent.querySelector( 'a' ).getAttribute( 'href' );
 				}
 
-				// Do not open the modal if we are looking at a gallery caption from before WP5, which may contain a link.
-				if ( parent.classList.contains( 'gallery-caption' ) ) {
+				// If the link does not point to the attachment or media file then assume Image has
+				// a custom link so don't load the carousel.
+				if (
+					parentHref &&
+					parentHref.split( '?' )[ 0 ] !==
+						target.getAttribute( 'data-orig-file' ).split( '?' )[ 0 ] &&
+					parentHref !== target.getAttribute( 'data-permalink' )
+				) {
 					return;
 				}
+			}
```

Comment:
This closing bracket looks like it is in the wrong place. It looks like the following two lines need to be inside this `if ( gallery )` code block, but you've shifted them out of the block:
```js
			var index = Array.prototype.indexOf.call( gallery.querySelectorAll( itemSelector ), item );
			loadSwiper( gallery, { startIndex: index } );
```

We can either move the closing bracket further down, or we can simply flip the opening condition so we have another early return instead of an indent:
```js
			if ( ! gallery ) {
				return;
			}

			if ( ! testForData( gallery ) ) {
				return;
			}
			// etc...
```

Reply:
Thanks, @daledupreez! That's been fixed [here](https://github.com/Automattic/jetpack/pull/25410/commits/1c7c315c8d381237fde700bcb8e4136a013a47e1).